docs/csiaddonsjob.md

```diff
@@ -0,0 +1,44 @@
+# CSIAddonsJob
+
+CSIAddonsJob is a namespaced custom resource designed to invoke operation to be performed on a volume.
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1beta1
+kind: CSIAddonsJob
+metadata:
+  name: sample-1
+spec:
+  operation:
+    reclaimSpace:
+      target:
+        pvc: pvc-1
+      parameters:
+        key1: value1
+        key2: value2
+  backOffLimit: 4
+  activeDeadlineSeconds: 500
+---
+apiVersion: csiaddons.openshift.io/v1beta1
+kind: CSIAddonsJob
+metadata:
+  name: sample-2
+spec:
+  operation:
+    reclaimSpace:
+      target:
+        pv: pvc-1232-4255-5675
```

Comment:
as this is a namespaced resource one validation we need to add is that PV is bound to a PVC that belongs to the same namespace? or remove this from spec for now and add it later when it's required.

Reply:
> as this is a namespaced resource one validation we need to add is that PV is bound to a PVC that belongs to the same namespace? 

Yes, the plan is to have some kind of validation to verify user can actually perform operation on that PV.

Reply:
drop PV now and add when there is a real requirement for it? we can always get to the PV if CR in a namespace has the corresponding PVC name.